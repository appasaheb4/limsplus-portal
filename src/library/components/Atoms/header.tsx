import React from "react";

const Header: React.FunctionComponent<any> = (props) => (
  <div className="sticky top-0 z-20 flex justify-between items-center px-2 py-2 border-b border-gray-300 bg-white">
    {props.children}
  </div>
);

export default Header;
