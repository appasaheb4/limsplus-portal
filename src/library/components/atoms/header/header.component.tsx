import React from 'react';

export const Header: React.FunctionComponent<any> = props => (
  <div className='sticky top-0 z-20 flex justify-between items-center px-2 py-2 border-b border-gray-300 bg-white'>
    {props.children}
  </div>
);

interface PageHeadingProps {
  title: string;
}

export const Heading: React.FunctionComponent<PageHeadingProps> = props => (
  <div className='text-center border-b border-gray-300 bg-white sticky top-14 z-20 font-FugazOne'>
    <h2 className='text-lg font-bold leading-4 text-gray-900 mt-0'>
      {props.title}
    </h2>
  </div>
);
interface PageHeadingProps {
  title: string;
  subTitle?: string;
}

export const PageHeading: React.FunctionComponent<PageHeadingProps> = props => (
  <div>
    <h2 className='text-lg font-bold leading-4 text-gray-900 mt-0'>
      {props.title}
    </h2>
    {props.subTitle && (
      <p className='text-xs leading-4 text-gray-500 m-0'>{props.subTitle}</p>
    )}
  </div>
);

interface PageHeadingLabDetailsProps {
  store: any;
}

export const PageHeadingLabDetails = (props: PageHeadingLabDetailsProps) => {
  const loginStore = props.store;
  return (
    <div className='flex flex-row items-center justify-between'>
      <img
        src={loginStore.login && loginStore.login.labLogo}
        alt='banner'
        className='object-fill h-10 w-12 rounded-md mr-2 d-none d-sm-inline-block'
      />
      <h2 className='text-base  leading-4 text-gray-900 mt-0'>
        {(loginStore.login &&
          loginStore.login.labList &&
          loginStore.login.labList.find(
            item => item.code === loginStore.login.lab,
          )?.name) ||
          loginStore.login?.lab}
      </h2>
    </div>
  );
};
