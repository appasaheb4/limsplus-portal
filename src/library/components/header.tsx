import React from "react";

const Header: React.FunctionComponent<any> = (props) => (
  <div className="sticky top-0 z-20 flex justify-between items-center px-4 py-3 border-b border-gray-200 bg-white">
    {props.children}
  </div>
);

export default Header;
