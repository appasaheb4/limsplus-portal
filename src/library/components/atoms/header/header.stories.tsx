import React from 'react';
import {ComponentStory, ComponentMeta} from '@storybook/react';
import {Header, PageHeading, PageHeadingLabDetails} from './header.component';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Header',
  component: Header,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: {control: 'color'},
  },
} as ComponentMeta<typeof Header>;

export const _Header: ComponentStory<typeof Header> = () => (
  <Header>
    <h1>Im Heading</h1>
  </Header>
);

export const _PageHeading: ComponentStory<typeof Header> = () => (
  <PageHeading title='Im Pagetitle' />
);

export const _PageHeadingLabDetails: ComponentStory<typeof Header> = () => (
  <PageHeadingLabDetails store='' />
);
