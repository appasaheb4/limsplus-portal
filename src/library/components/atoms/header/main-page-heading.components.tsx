import React from 'react';
import { Header, PageHeading, PageHeadingLabDetails } from './header.component';
import { connect } from 'react-redux';
import { useStores } from '@/stores';
import { useHistory } from 'react-router-dom';
import { RouterFlow } from '@/flows';
import { AutocompleteSearch } from '../../molecules/auto-complete-search/autocomplete-search.component';

const MainPageHeading = ({ sidebar, title, store }) => {
  const { routerStore } = useStores();
  const history = useHistory();
  return (
    <div className='relative'>
      <Header>
        <PageHeading title={title || ''} />
        <PageHeadingLabDetails store={store} />
      </Header>

      <div className='absolute left-[50%] top-7 transform -translate-x-1/2 -translate-y-1/2 w-[500px]  z-99'>
        {!sidebar?.isOpen && (
          <AutocompleteSearch
            data={routerStore.userRouter}
            onChange={async (item: any, children: any) => {
              const { permission, selectedComp } =
                await RouterFlow.updateSelectedCategory(
                  item?.name,
                  children?.name,
                );
              routerStore.updateSelectedComponents(selectedComp);
              routerStore.updateUserPermission(permission);
              history.replace(children.path);
            }}
          />
        )}
      </div>
    </div>
  );
};

export default connect((store: any) => ({
  sidebar: store.sidebar,
}))(MainPageHeading);
