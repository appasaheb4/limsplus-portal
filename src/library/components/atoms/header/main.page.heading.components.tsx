import React from 'react';
import { Header, PageHeading, PageHeadingLabDetails } from './header.component';
import { connect } from 'react-redux';
import { useStores } from '@/stores';
import { useHistory } from 'react-router-dom';
import { RouterFlow } from '@/flows';
import { AutocompleteSearch } from '../../molecules/auto-complete-search/autocomplete-search.component';

const MainPageHeading = ({ sidebar, title, store }) => {
  const { routerStore } = useStores();
  const history = useHistory();
  return (
    <div className=' relative'>
      <Header>
        <PageHeading title={title || ''} />
        {!sidebar?.isOpen && (
          <div style={{ width: '50%' }}>
            <AutocompleteSearch
              data={routerStore.userRouter}
              onChange={async (item: any, children: any) => {
                const { permission, selectedComp } =
                  await RouterFlow.updateSelectedCategory(
                    item?.name,
                    children?.name,
                  );
                routerStore.updateSelectedComponents(selectedComp);
                routerStore.updateUserPermission(permission);
                history.replace(children.path);
              }}
            />
          </div>
        )}
        <PageHeadingLabDetails store={store} />
      </Header>
    </div>
  );
};

export default connect((store: any) => ({
  sidebar: store.sidebar,
}))(MainPageHeading);
