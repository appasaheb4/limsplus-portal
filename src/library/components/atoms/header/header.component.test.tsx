import React from 'react';
import {render} from '@utils';
import {Header, PageHeading, PageHeadingLabDetails} from './header.component';

it('render header correctly', () => {
  const header = render(<Header> Im Heading</Header>);
  expect(header).toMatchSnapshot();
});

it('render pageHeading correctly', () => {
  const pageHeading = render(<PageHeading title="Im Page Title" />);
  expect(pageHeading).toMatchSnapshot();
});
it('render pageHeadingLabDetails correctly', () => {
  const pageHeadingLabDetails = render(<PageHeadingLabDetails store={''} />);
  expect(pageHeadingLabDetails).toMatchSnapshot();
});
