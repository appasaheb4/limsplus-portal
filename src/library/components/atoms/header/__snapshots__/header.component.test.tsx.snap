// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render header correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sticky top-0 z-20 flex justify-between items-center px-2 py-2 border-b border-gray-300 bg-white"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sticky top-0 z-20 flex justify-between items-center px-2 py-2 border-b border-gray-300 bg-white"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Heading component render imagebackground correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center border-b border-gray-300 bg-white sticky top-14 z-20 font-FugazOne"
      >
        <h2
          class="text-lg font-bold leading-4 text-gray-900 mt-0"
        >
          frefg
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center border-b border-gray-300 bg-white sticky top-14 z-20 font-FugazOne"
    >
      <h2
        class="text-lg font-bold leading-4 text-gray-900 mt-0"
      >
        frefg
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PageHeading component render pageHeading correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-lg font-bold leading-4 text-gray-900 mt-0"
        >
          frefg
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-lg font-bold leading-4 text-gray-900 mt-0"
      >
        frefg
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PageHeadingLabDetails component render pageHeadingLabDetails correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row items-center"
      >
        <img
          alt="banner"
          class="object-fill h-10 w-12 rounded-md mr-2"
        />
        <h2
          class="text-base  leading-4 text-gray-900 mt-0"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row items-center"
    >
      <img
        alt="banner"
        class="object-fill h-10 w-12 rounded-md mr-2"
      />
      <h2
        class="text-base  leading-4 text-gray-900 mt-0"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
